src/metadatacache/MetadataCache.ts

```diff
@@ -218,6 +218,61 @@ export class MetadataCache {
   }
 
   /**
+   * Edit Metadata for a DDO.
+   * @param  {did} string DID.
+   * @param  {newMetadata}  EditableMetadata Metadata fields & new values.
+   * @param  {String} updated Updated field of the DDO
+   * @param  {String} signature Signature using updated field to verify that the consumer has rights
+   * @return {Promise<String>} Result.
+   */
+  public async editMetadata(
```

Comment:
so there has been a drastic change, this is now basically in `ocean.assets.editMetadata()` but it only does an object restructuring https://github.com/oceanprotocol/ocean.js/blob/main/src/ocean/Assets.ts#L253

and then the real update is done with `ocean.assets.updateMetadata()` https://github.com/oceanprotocol/ocean.js/blob/main/src/ocean/Assets.ts#L275

super confusing naming and all but if we want to validate anything it needs to happen over there

Reply:
>> if we want to validate anything it needs to happen over there

You mean in the `ocean.assets.updateMetadata()` ?